packages/core/user-settings/core-user-settings-server-internal/user_settings_service.test.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { mockCoreContext } from '@kbn/core-base-server-mocks';
+import { UserSettingsService } from './user_settings_service';
+import { KibanaRequest } from '@kbn/core-http-server';
+
+describe('#setup', () => {
+  const coreContext: ReturnType<typeof mockCoreContext.create> = mockCoreContext.create();
+
+  it('fetches userSettings when client is set and returns proper values', async () => {
+    const service = new UserSettingsService(coreContext);
+    const { setUserProfileSettings, getUserSettingDarkMode } = service.setup();
+
+    setUserProfileSettings({
+      get: jest.fn().mockReturnValueOnce(Promise.resolve({ darkMode: 'dark' })),
+    });
+
+    const darkMode = await getUserSettingDarkMode({} as KibanaRequest);
+    expect(darkMode).toEqual('dark');
```

Comment:
NIT: I'll assert that the `UserProfileSettingsClientContract.get` method was called with the proper parameters

```ts
const userProfileContract = {
      get: jest.fn().mockReturnValueOnce(Promise.resolve({ darkMode: 'dark' })),
    }
setUserProfileSettings(userProfileContract);

const request =  HttpServiceMock.createRequest(); // don't remember the exact name of the mock for this
const darkMode = await getUserSettingDarkMode(request);

expect(darkMode).toEqual('dark');
expect(userProfileContract.get).toHaveBeenCalledTimes(1);
expect(userProfileContract.get).toHaveBeenCalledWith(request)
```

Reply:
Nice - Ive been looking for a nice way to mock KibanaRequest! Fixed in latest commit.

*These tests were also updated for the new `boolean | undefined` return value